spec/unit/analytics/ga4_import/relevance_report_generator_spec.rb

```diff
@@ -0,0 +1,40 @@
+require "spec_helper"
+
+RSpec.describe Analytics::Ga4Import::RelevanceReportGenerator do
+  let(:data_fetcher) { instance_double(Analytics::Ga4Import::DataFetcher) }
+  let(:page_view_consolidator) { instance_double(Analytics::Ga4Import::PageViewConsolidator) }
+  let(:relevancy_calculator) { instance_double(Analytics::Ga4Import::ElasticSearchRelevancySerialiser) }
+
+  before do
+    stub_classes
```

Comment:
```suggestion
    allow(Analytics::Ga4Import::DataFetcher).to receive(:new).and_return(data_fetcher)
    allow(data_fetcher).to receive(:call).and_return(:paginated_data)

    allow(Analytics::Ga4Import::PageViewConsolidator).to receive(:new).and_return(page_view_consolidator)
    allow(page_view_consolidator).to receive(:consolidated_page_views).and_return(:consolidated_page_views)

    allow(Analytics::Ga4Import::ElasticSearchRelevancySerialiser).to receive(:new).and_return(relevancy_calculator)
    allow(relevancy_calculator).to receive(:relevance).and_return(:relevance)
```

It seems unnecessary to for these to be in a separate method